code/src/components/app-dropdown/AppDropdown.tsx

```diff
@@ -0,0 +1,84 @@
+import { useState, useRef, MouseEvent, FocusEvent, ReactNode } from "react";
+import KeyboardArrowDownIcon from "@mui/icons-material/KeyboardArrowDown";
+import KeyboardArrowUpIcon from "@mui/icons-material/KeyboardArrowUp";
+import AppBox from "@/components/app-box/AppBox";
+import AppTypography from "@/components/app-typography/AppTypography";
+import AppDropdownOptions from "@/components/app-dropdown/components/AppDropdownOptions";
+import { AppDropdownProps } from "@/components/app-dropdown/AppDropdown.types";
+
+import "@/components/app-dropdown/AppDropdown.scss";
+
+const AppDropdown = ({
+  options,
+  onSelect,
+  defaultLabel,
+  className
+}: AppDropdownProps) => {
+  const selectedLabelValue = defaultLabel || options[0].label;
+
+  const [isOpen, setIsOpen] = useState(false);
+  const [selectedLabel, setSelectedLabel] =
+    useState<ReactNode>(selectedLabelValue);
+  const dropdownRef = useRef<HTMLElement>(null);
+
+  const iconConditional = isOpen ? (
+    <KeyboardArrowUpIcon fontSize="medium" />
+  ) : (
+    <KeyboardArrowDownIcon fontSize="medium" />
+  );
+
+  const handleSelect = (value: string, label: ReactNode) => {
+    setSelectedLabel(label);
+    setIsOpen(false);
+    onSelect(value);
+  };
+
+  const handleToggleDropdownClick = () => {
+    setIsOpen((prevOpenState) => !prevOpenState);
+  };
+
+  const handleOptionClick = (
+    event: MouseEvent,
+    value: string,
+    label: ReactNode
+  ) => {
+    event.stopPropagation();
```

Comment:
Why did you use `event.stopPropagation();` here?

Reply:
Can you explain in more detail?

Reply:
> Can you explain in more detail?

Sure. It's propagate to the parent element `AppBox` in my case which has `onClick` event handler. It's would inadvertently toggle dropdown causing it will not close immediately after choosing sort variant opening which is not the intended behavior
using `event.stopPropagation()` the component behaves as expected


https://github.com/idx-academy/spa-orders/assets/56305508/e9be63a1-333d-422a-a884-6f35caccffa0

Reply:
@mhevyk also I have a good article about it) 
https://developer.mozilla.org/en-US/docs/Web/API/Event/stopPropagation

Reply:
Thanks for useful explanation. You actually helped me to understand how it really works